lib/plugin.js

```diff
@@ -32,14 +32,16 @@ function gtmServer(ctx, initialized) {
       return
     }
 
-    const gtmScript = ctx.app.head.script.find(s => s.hid == '<%= options.scriptId %>')
+    const head = typeof ctx.app.head === 'function' ? ctx.app.head() : ctx.app.head;
+
+    const gtmScript = head.script.find(s => s.hid == '<%= options.scriptId %>')
     gtmScript.innerHTML = `window['${_layer}']=${JSON.stringify(events)};${gtmScript.innerHTML}`
 
     if (inits.length) {
       gtmScript.innerHTML += `;${JSON.stringify(inits)}.forEach(function(i){window._gtm_inject(i)})`
     }
 <% if (options.noscript) { %>
-    const gtmIframe = ctx.app.head.noscript.find(s => s.hid == '<%= options.noscriptId %>')
+    const gtmIframe = head.noscript.find(s => s.hid == '<%= options.noscriptId %>')
```

Comment:
This is not a correct fix. The intention of the original code was to modify the `innerHTML` of the script added to `head`. When `head` is a function you can't modify the original script as your will (most likely) receive a new head object every time the function is called.

So this change will prevent the crash while at the same time make this part of code not do anything useful.